components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/test/java/org/wso2/carbon/identity/role/v2/mgt/core/dao/RoleDAOTest.java

```diff
@@ -376,6 +377,29 @@ public void testGetPermissionListOfRole() throws Exception {
         Assert.assertEquals(getPermissionNameList(rolePermissions), getPermissionNameList(permissions));
     }
 
+    @Test
+    public void testGetPermissionListOfRoles() throws Exception {
```

Comment:
This test is not covering shared role's permission list retrieval

Reply:
Added an additional test for the shared role permission extraction